src/toast/cc-toast.js

```diff
@@ -0,0 +1,339 @@
+import { css, html, LitElement } from 'lit-element';
+import { classMap } from 'lit-html/directives/class-map.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { dispatchCustomEvent } from '../lib/events.js';
+import { i18n } from '../lib/i18n.js';
+import { defaultThemeStyles } from '../styles/default-theme.js';
+
+const closeSvg = new URL('../assets/close-gray.svg', import.meta.url).href;
+const infoSvg = new URL('../assets/information-line.svg', import.meta.url).href;
+const successSvg = new URL('../assets/checkbox-circle-line.svg', import.meta.url).href;
+const warningSvg = new URL('../assets/alert-line.svg', import.meta.url).href;
+const dangerSvg = new URL('../assets/spam-2-line.svg', import.meta.url).href;
+
+/**
+ * @typedef {import('./types.js').NotificationIntent} NotificationIntent
+ */
+
+/**
+ * A component dedicated to display a short message and dismiss it after a short period of time.
+ *
+ * ## Details
+ *
+ * This component is not to be used directly but should be used through the `cc-toaster` component.
+ *
+ * * The component displays a message. Optionally, this message can be topped by a heading.
+ * * The component dispatches a `cc-toast:dismiss` event after a certain amount of time. The timer starts when the component is attached to the DOM.
+ * * Optionally, it can be closed manually with a close button.
```

Comment:
I think we should be more explicit on the fact that clicking on the close button will also trigger a `cc-toast:dismiss` event.